wandb/sdk/artifacts/artifact.py

```diff
@@ -277,67 +277,72 @@ def _from_attrs(
         # Placeholder is required to skip validation.
         artifact = cls("placeholder", type="placeholder")
         artifact._client = client
-        artifact._id = attrs["id"]
         artifact._entity = entity
         artifact._project = project
         artifact._name = name
+        artifact._assign_attrs(attrs)
+        return artifact
+
+    def _assign_attrs(self, attrs: dict[str, Any]) -> None:
```

Comment:
note: ensures that `Artifact._from_attrs` and `Artifact._update` share the same parsing + assignment code, on receiving the JSON artifact obj from a GQL response